### PR TITLE
auto tz updates in GHA

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
# Issue
Extends #4510 by checking out the new tz release, possibly updating the URL used in `scripts/valhalla_build_elevation`, committing and ppushing everything and opening a new PR. Not very experienced with using the GitHub CLI inside GHA, but if `GITHUB_TOKEN` has the necessary privileges, we should be able to pull this off.   

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: .github/workflows/check_tz_releases.yml [hunk 1]
```diff
@@ -25,7 +25,46 @@ jobs:
...
+              git add third_party/tz
+
+              # update the url in valhalla_build_timezones 
+              url_old=$(grep -oP "url=\"\Khttps://github.com/.*(?=\")" scripts/valhalla_build_timezones)
+              url_new=$(echo ${url_old} | sed -E "s/download\/[0-9]{4}[a-z]/download\/${latest_tag}/")
+
+              # first make sure the file is actually available
+
+              set +e
+              curl -Is --fail-with-body ${url_new} >/dev/null
```
Comment (chrstnbwnkl) [inline .github/workflows/check_tz_releases.yml:42]: check if the resource exists

File: .github/workflows/check_tz_releases.yml [hunk 2]
```diff
@@ -25,7 +25,46 @@ jobs:
...
+
+              if [[ $? -eq 0 ]]; then
```
Comment (chrstnbwnkl) [inline .github/workflows/check_tz_releases.yml:44]: if so, it's safe to replace the url in the script. Just need to replace the old tag with the new one.

File: .github/workflows/check_tz_releases.yml [hunk 3]
```diff
@@ -25,7 +25,46 @@ jobs:
...
+                  git diff scripts/valhalla_build_timezones
+              fi
+
+              set -e
+              # commit and push 
+              git commit -m "bumped tz to ${latest_tag}"
+              git push origin 
+
+              # open new PR 
+              gh pr create --base master --head $new_branch --title "Bump tz to ${latest_tag}" --body "Created by workflow run [#${WORKFLOW_RUN_ID}](https://github.com/valhalla/valhalla/actions/runs/$WORKFLOW_RUN_ID/job/${WORKFLOW_JOB_ID})"
```
Comment (kevinkreiser) [inline .github/workflows/check_tz_releases.yml:61]: `gh pr create` what is this magic command!? i never knew there was a command line github utility, cool!

Comment (nilsnolde) [reply .github/workflows/check_tz_releases.yml:61]: Should we add the link to the tz commit’s NEWS doc in the PR description so we can conveniently check what changed?

**Review by nilsnolde — COMMENTED** [review]
Looks great, thanks! 

Let’s also add a “workflow_dispatch” so we can trigger manually?

**Review by nilsnolde — APPROVED** [review]
Cool thanks. Let’s merge and trigger to see the magic unfold :)

**Comment by nilsnolde** [comment]
Yeah there’s Auth problems with git. Was hoping it’ll know from context, but it doesn’t. Should we try on a fork first @chrstnbwnkl ?

**Comment by nilsnolde** [comment]
Seems quite easy though: https://stackoverflow.com/a/58393457. Is there a valhalla bot user with repo access @kevinkreiser? If not, might make sense to create one for automations like these?

**Comment by chrstnbwnkl** [comment]
Ah my assumption was it would automatically assume some sort of GHA bot identity :smile:

**Comment by chrstnbwnkl** [comment]
The `GITHUB_TOKEN` permissions also need to include write, which I don't think is enabled by default.

**Pull request merged**